tests/acceptance/stepDefinitions/sharingContext.js

```diff
@@ -47,41 +48,73 @@ const userSharesFileOrFolderWithUser = function (file, sharee, role) {
 const userSharesFileOrFolderWithGroup = function (file, sharee, role) {
   return userSharesFileOrFolderWithUserOrGroup(file, sharee, true, role)
 }
-
 /**
+ * creates a new share
  *
  * @param {string} elementToShare  path of file/folder being shared
  * @param {string} sharer  username of the sharer
- * @param {string} receiver  username of the receiver
- * @param {number} shareType  Type of share 0 = user, 1 = group, 3 = public (link), 6 = federated (cloud share).
- * @param {string} permissions  permissions of the share for valid permissions see sharingHelper.PERMISSION_TYPES
+ * @param receiver  username of the receiver
+ * @param shareType  Type of share 0 = user, 1 = group, 3 = public (link), 6 = federated (cloud share).
+ * @param {string} permissionString  permissions of the share for valid permissions see sharingHelper.PERMISSION_TYPES
+ * @param {string} name name of the link (for public links), default = "New Share"
+ * @param {object} extraParams Extra parameters allowed on the share
+ * @param {string} extraParams.password Password of the share (public links)
+ * @param {string} extraParams.expireDate Expiry date of the share
  */
-const shareFileFolder = function (elementToShare, sharer, receiver, shareType = SHARE_TYPES.user, permissionString = 'all', password = null) {
-  const permissions = sharingHelper.humanReadablePermissionsToBitmask(permissionString)
+const shareFileFolder = function (
+  elementToShare, sharer, receiver = null, shareType = SHARE_TYPES.user,
+  permissionString = 'all', name = 'New Share', extraParams = {}
```

Comment:
don't have a default name. If no name is given, create the share without a name

Reply:
:rocket: